diff --git a/composer.json b/composer.json
index 943e313..81eb9f8 100644
--- a/composer.json
+++ b/composer.json
@@ -7,6 +7,9 @@
         "psr-4": { "": "src/" },
         "files": ["app/AppKernel.php"]
     },
+    "autoload-dev": {
+        "psr-4": { "Tests\\": "tests/" }
+    },
     "require": {
         "php"                                  : ">=5.3.9",
         "ext-pdo_sqlite"                       : "*",
diff --git a/app/phpunit.xml.dist b/phpunit.xml.dist
similarity index 100%
rename from app/phpunit.xml.dist
rename to phpunit.xml.dist
diff --git a/src/AppBundle/Tests/Controller/Admin/BlogControllerTest.php b/tests/AppBundle/Controller/Admin/BlogControllerTest.php
similarity index 98%
rename from src/AppBundle/Tests/Controller/Admin/BlogControllerTest.php
rename to tests/AppBundle/Controller/Admin/BlogControllerTest.php
index 873ecc7..ae906dc 100644
--- a/src/AppBundle/Tests/Controller/Admin/BlogControllerTest.php
+++ b/tests/AppBundle/Controller/Admin/BlogControllerTest.php
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace AppBundle\Tests\Controller\Admin;
+namespace Tests\AppBundle\Controller\Admin;
 
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 use Symfony\Component\HttpFoundation\Response;
diff --git a/src/AppBundle/Tests/Controller/BlogControllerTest.php b/tests/AppBundle/Controller/BlogControllerTest.php
similarity index 96%
rename from src/AppBundle/Tests/Controller/BlogControllerTest.php
rename to tests/AppBundle/Controller/BlogControllerTest.php
index 43c5aa7..793a169 100644
--- a/src/AppBundle/Tests/Controller/BlogControllerTest.php
+++ b/tests/AppBundle/Controller/BlogControllerTest.php
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace AppBundle\Tests\Controller;
+namespace Tests\AppBundle\Controller;
 
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 use AppBundle\Entity\Post;
diff --git a/src/AppBundle/Tests/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
similarity index 98%
rename from src/AppBundle/Tests/Controller/DefaultControllerTest.php
rename to tests/AppBundle/Controller/DefaultControllerTest.php
index 24f6812..5eea3ff 100644
--- a/src/AppBundle/Tests/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace AppBundle\Tests\Controller;
+namespace Tests\AppBundle\Controller;
 
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 
diff --git a/src/AppBundle/Tests/Utils/SluggerTest.php b/tests/AppBundle/Utils/SluggerTest.php
similarity index 97%
rename from src/AppBundle/Tests/Utils/SluggerTest.php
rename to tests/AppBundle/Utils/SluggerTest.php
index 6f67724..b3baa44 100644
--- a/src/AppBundle/Tests/Utils/SluggerTest.php
+++ b/tests/AppBundle/Utils/SluggerTest.php
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace Tests\Utils;
+namespace Tests\AppBundle\Utils;
 
 use AppBundle\Utils\Slugger;
 
