diff --git a/app/console b/bin/console
similarity index 94%
rename from app/console
rename to bin/console
index b405752..d0b3948 100755
--- a/app/console
+++ b/bin/console
@@ -7,7 +7,7 @@
 
 set_time_limit(0);
 
-require_once __DIR__.'/bootstrap.php.cache';
+require __DIR__.'/../app/autoload.php';
 
 use Symfony\Bundle\FrameworkBundle\Console\Application;
 use Symfony\Component\Console\Input\ArgvInput;
diff --git a/composer.json b/composer.json
index 56a7252..7272038 100644
--- a/composer.json
+++ b/composer.json
@@ -52,8 +52,7 @@
     "config": {
         "platform": {
             "php": "5.3.9"
-        },
-        "bin-dir": "bin"
+        }
     },
     "extra": {
         "symfony-app-dir": "app",
